src/main/java/org/liquidengine/legui/layout/LayoutManager.java

```diff
@@ -0,0 +1,40 @@
+package org.liquidengine.legui.layout;
+
+import org.liquidengine.legui.component.Frame;
+
+/**
+ * Layout manager. Used to layout component and it's child components.
+ * @author Aliaksandr_Shcherbin.
+ */
+public abstract class LayoutManager {
+
+    /**
+     * Instance of layout manager.
+     */
+    private static LayoutManager instance = new DefaultLayoutManager();
+
+    /**
+     * Returns layout manager instance.
+     *
+     * @return layout manager instance.
+     */
+    public static LayoutManager getInstance() {
+        return instance;
+    }
+
+    /**
+     * Used to set layout manager instance.
+     *
+     * @param instance layout manager instance to set.
+     */
+    public static void setInstance(LayoutManager instance) {
+        LayoutManager.instance = instance;
+    }
+
+    /**
+     * Used to layout frame layers and all of their child components.
+     *
+     * @param frame frame to lay out.
+     */
+    public abstract void layout(Frame frame);
```

Comment:
Same as above

Reply:
Not really clear what is your question here.

Reply:
I was trying to say that this line was refering to the other comment: `Why this method is not in LayoutManager? can you only layout Frames? Since the interface Layout allow to modify components the LayoutManager should be able to modify only a subtree of components instead of all the components on a Frame` but you already responded to it